### PR TITLE
Enhancement/schedule page/time and search occupied

### LINKED ISSUE #8757: Reduce area occupied by time and search field on schedule page
On the mobile view the schedule page fields for time and search occupy two rows. Please reduce the required space and show both fields in one row. When the user clicks on the search field automatically expand the width of the search field to cover the entire row.

Current: **This is the current view:**
Screenshot from 2023-05-26 17-45-12
compare https://eventyay.com/e/7cfe0771/schedule

Expected: **Both fields should be there still, but when the user clicks into the search box it should expand.**

Intended view Screenshot from 2023-07-03 12-40-51a

Compare the idea of an expanding search box here for example https://codepen.io/k185/pen/PQajXE

### PR DESCRIPTION
**Pull request by blackmouse572**
Resolves  #8757 Reduce area occupied by time and search field on schedule page

#### Short description of what this resolves:
- Change Seach box behaviour
- Search and time occupied 1 instead of 2 rows

#### Changes proposed in this pull request:

- Search Box Input which will take whole area to shorter version of it. When use focus, it will expand to original.
- Search Box and Timezone place in 1 row on mobile screen instead of 2. These components is space between each others

**BEFORE**
image

**AFTER**
image

-- On Click
image

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Thank you. Nice work. Please see comments below.

1. On your screenshot it is already visible that the height of the mobile time and search boxes is different. The time box and search box should have the same height and there should be a space between them.
2. The search box should expand over the time box when clicked.
Current PR
Screenshot from 2023-07-03 12-40-51

Intended view
Screenshot from 2023-07-03 12-40-51a

Compare the idea of an expanding search box here for example https://codepen.io/k185/pen/PQajXE

**Review by cweitat — CHANGES_REQUESTED** [review]
https://github.com/fossasia/open-event-frontend/pull/8893#issuecomment-1617901591

**Comment by blackmouse572** [comment]
> Thank you. Nice work. Please see comments below.
> 
> 1. On your screenshot it is already visible that the height of the mobile time and search boxes is different. The time box and search box should have the same height and there should be a space between them.
> 2. The search box should expand over the time box when clicked.
>    Current PR
>    Screenshot from 2023-07-03 12-40-51
> 
> Intended view Screenshot from 2023-07-03 12-40-51a
> 
> Compare the idea of an expanding search box here for example https://codepen.io/k185/pen/PQajXE
> 
> Test at https://eventyay.com/e/7cfe0771/schedule and https://open-event-frontend-git-fork-blackmouse572-enha-54d351-eventyay.vercel.app/e/7cfe0771/schedule

I have change based on the reply. this is the newest version. 
image

- I have change from default SearchBox into conditional render. Whether it is mobile device to implement the custom scss expand search box. This maybe cause the boilerplate but it easy to maintain and modified.

**Comment by cweitat** [comment]
image
@blackmouse572 uneven height

File: app/styles/libs/_helpers.scss [hunk 1]
```diff
@@ -1,11 +1,13 @@
 $font-weights: 100 200 300 400 500 600 700 800 900;
+
```
Comment (cweitat) [inline app/styles/libs/_helpers.scss:2]: do not include unnecessary line breaks

File: app/styles/libs/_helpers.scss [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 @each $font-weight in $font-weights {
   .weight-#{$font-weight} {
     font-weight: $font-weight !important;
   }
 }
 
 $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
+
```
Comment (cweitat) [inline app/styles/libs/_helpers.scss:10]: same for this and rest

File: app/styles/pages/public-session.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
+.session-search-box {
+    width: 100%;
```
Comment (cweitat) [inline app/styles/pages/public-session.scss:2]: Github Lint

File: app/styles/pages/public-session.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    .search-box {
```
Comment (cweitat) [inline app/styles/pages/public-session.scss:4]: and the rest

**Review by cweitat — CHANGES_REQUESTED** [review]

**Comment by cweitat** [comment]
Please include new screenshot of latest update

**Comment by blackmouse572** [comment]
--EMPTY STATE
image
--FOCUS STATE
image

**Review by cweitat — CHANGES_REQUESTED** [review]
uncompliance of formatting

**Comment by cweitat** [comment]
image

**Comment by cweitat** [comment]
image

**Comment by blackmouse572** [comment]
Sorry for the late, just fix the scss break point to display gap on mobile device, also move the `transition` to input selector for the animation.

**Comment by cweitat** [comment]
@blackmouse572 is this considered as fixed? do fix your codacy too

**Comment by blackmouse572** [comment]
> @blackmouse572 is this considered as fixed? do fix your codacy too

Yes, really sorry for my latency, I am working on it. Secondly, I notice from the tablet screen, the gap is not apply, so i change the breakpoint to suitable value so that when the search and timezone apply same row, the gap is applied to. That why i consider a fix. If any problem please let me know. Thank you for you time <3

**Comment by mariobehling** [comment]
Thank you for your contribution. I am closing this PR as it has been solved in the meantime.

**Pull request closed without merging**